avaframe/com1DFA/com1DFA.py

```diff
@@ -51,42 +52,79 @@
-        cfgFilesRels: list
-            list of paths to one cfgFile for each releaseScenario with updated thickness values
-
+        simDict: dict
+            dictionary with one key per simulation to perform including its config object
+        inputSimFiles: dict
+            dictionary with input files info
+        outDir: str
+            path to store outputs
     """
 
-    # check if thickness settings in ini file are valid
-    for thType in ['entTh', 'relTh', 'secondaryRelTh']:
-        _ = dP.checkThicknessSettings(cfgInitial, thType)
+    modName = 'com1DFA'
 
-    # fetch input data - dem, release-, entrainment- and resistance areas (and secondary release areas)
-    inputSimFilesAll = gI.getInputDataCom1DFA(avaDir, cfgInitial)
+    # read initial configuration
+    if isinstance(cfgInfo, (pathlib.Path, str)) or cfgInfo == '':
+        cfgStart = cfgUtils.getModuleConfig(com1DFA, fileOverride=cfgInfo, toPrint=False)
+    elif isinstance(cfgInfo, configparser.ConfigParser):
+        cfgStart = cfgInfo
 
-    # get thickness of release and entrainment areas (and secondary release areas) -if thFromShp = True
-    inputSimFilesAll, cfgFilesRels = gI.getThickness(inputSimFilesAll, avaDir, modName, cfgInitial)
+    # Create output and work directories
+    workDir, outDir = inDirs.initialiseRunDirs(avalancheDir, modName,
+                                               cfgStart['GENERAL'].getboolean('cleanDEMremeshed'))
 
-    return inputSimFilesAll, cfgFilesRels
+    # read input data files (release, DEM, etc.) and update configuration with this
+    inputSimFilesAll, cfgStart = com1DFATools.setRelThIni(avalancheDir, com1DFA, cfgStart)
 
+    # initialise reportDictList and flag indicating whether simulations have been performed
+    reportDictList = []
+    simsPerformed = False
+    #
+    # # copy inputSimFilesAll - as this becomes changed according to current release scenario
+    # inputSimFiles = inputSimFilesAll.copy()
 
-def com1DFAMain(avalancheDir, cfgMain, cfgFile=''):
+    # reset variationDict
+    variationDict = ''
+
+    # create a dictionary with information on which parameter shall be varied for individual simulations
+    # compare cfgStart to default module config for this
+    modCfg, variationDict = dP.getParameterVariationInfo(avalancheDir, com1DFA, cfgStart)
+
+    # first fetch info on already existing simulations in Outputs
+    # if need to reproduce exactly the hash - need to be strings with exactely the same number of digits!!
+    simDFOld, simNameOld = cfgUtils.readAllConfigurationInfo(avalancheDir, specDir='')
+
+    # create a configuration object per simulation to run (from configuration) gathered in simDict
+    # only new simulations are included in this simDict
+    # key is simName and corresponds to one simulation
+    simDict = prepareVarSimDict(modCfg, inputSimFilesAll, variationDict, simNameOld=simNameOld)
```

Comment:
why do we have this as a dict and not a DF?
you use simDFOld, why not also simDFNew?

Reply:
it is a dictionary as it has different types of objects for the keys, the configparser object, strings, floats, ... simDFOld was introduced to distinguish which simulations are already present in the outputs, simDF is the one for the actual simulation run - could be renamed to new, but could stay just simDF too I think

Reply:
rename to exisitng